src/mon/MonClient.h

```diff
@@ -213,14 +213,19 @@ class MonClient : public Dispatcher {
   void handle_subscribe_ack(MMonSubscribeAck* m);
 
   bool _sub_want(string what, version_t start, unsigned flags) {
-    if ((sub_new.count(what) == 0 &&
-	 sub_sent.count(what) &&
-	 sub_sent[what].start == start &&
-	 sub_sent[what].flags == flags) ||
-	(sub_new.count(what) &&
-	 sub_new[what].start == start &&
-	 sub_new[what].flags == flags))
+    auto sub_new_iter = sub_new.find(what);
```

Comment:
with the haystack_vector example I would use 
`auto haystack_iter = haystack_vector.find(straw_i);`
which for me is much more readeable than 
`auto found = v.find(straw);`

haystack_iter tells me straight away that I've got a variable pointing to an element in haystack that is an iterator, so I have to dereference it if I want to access the value.
When I look at a variable named `found` I think about boolean type. 

I have changed it to 'sub' as you proposed earlier, but I think it's now less readable than it was before:) For me at first glance it looks like we are comparing the same thing in both branches of if...